openmdao/components/indep_var_comp.py

```diff
@@ -23,6 +23,9 @@ class IndepVarComp(Component):
     def __init__(self, name, val=None, **kwargs):
         super(IndepVarComp, self).__init__()
 
+        if 'promotes' in kwargs:
+            raise ValueError('IndepVarComp init: promotes is not supported in IndepVarComp.')
+
```

Comment:
Do you mean here in system.py?  Should I add a **kwargs to __init__ ?

class System(object):
    """ Base class for systems in OpenMDAO. When building models, user should
    inherit from `Group` or `Component`
    """

    def __init__(self):